hibike/devices/BatteryBuzzer/voltage_sense.cpp

```diff
@@ -0,0 +1,212 @@
+#include "voltage_sense.h"
+
+//code that deals with measuring cell voltages.
+//also deals with calibration.
+
+int cell1 = A0;
+int cell2 = A1;
+int cell3 = A2;
+
+int enable = 16;
+int calib_button = 21;
+
+
+
+void setup_sensing()
+{
+
+  pinMode(enable, OUTPUT);
+  pinMode(calib_button, INPUT);
+
+  
+
+  //for stable (if unknown) internal voltage.
+  analogReference(INTERNAL);
+
+  //let's turn enable on, so we can measure the battery voltage.
+  digitalWrite(enable, HIGH);
+
+  if(triple_calibration)
+  {
+    update_triple_calibration();  //and then update the calibration values I use based on what I now know
+  }
+  else
+  {
+    float val = get_calibration();
+    if(val > 0) //by convention, val is -1 if there's no calibration.
+    {
+      vref_guess = val;   
+    }
+  }
+  
+}
+
+
+
+void measure_cells() //measures the battery cells. Should call at least twice a second, but preferrably 5x a second.  No use faster.
+{
+    int r_cell1 = analogRead(cell1);
+    int r_cell2 = analogRead(cell2);
+    int r_cell3 = analogRead(cell3);
+
+    //Serial.println("Raw Voltages");
+    //Serial.println(5*float(r_cell1)/1024);
+    //Serial.println(5*float(r_cell2)/1024);
+    //Serial.println(5*float(r_cell3)/1024);
+
+
+    v_cell1 = float(r_cell1) * (10 + 10.0) / 10.0 * calib[0] / 1024;  // (10k + 10k)/(10k) to extract cell voltage in counts, then convert 1024 counts per vref volts.
+    v_cell2 = float(r_cell2) * (30.0 + 10) / 10.0 * calib[1] / 1024;  // (30k + 10k)/(10k) to extract cell voltage in counts, then convert 1024 counts per vref volts.
+    v_cell3 = float(r_cell3) * (51.00 + 10.0 ) / 10.0 * calib[2] / 1024;  // (51k + 10k)/(10k) to extract cell voltage in counts, then convert 1024 counts per vref volts.
+  
+    dv_cell2 = v_cell2 - v_cell1;
+    dv_cell3 = v_cell3 - v_cell2;
+
+    v_batt = v_cell3;
+
+    // if(prints)
+    // {
+    //   String str_cell1 = String(v_cell1, 3);
+    //   String str_cell2 = String(v_cell2, 3);
+    //   String str_cell3 = String(v_cell3, 3);
+  
+  
+  
+    //   String sdv_cell2 = String(dv_cell2, 3);
+    //   String sdv_cell3 = String(dv_cell3, 3);
+  
+
+    //   if(triple_calibration)
+    //   {
+    //     Serial.println("Triple Calibration Mode");
+    //   }
+    //   else
+    //   {
+    //     Serial.println("Single Calibration Mode");
+    //   }
+  
+    //   Serial.println("ABS: Cell1 : " + str_cell1 + " Cell2 : " + str_cell2 + " Cell3 : " + str_cell3);
+    //   Serial.println("REL: Cell1: " + str_cell1 + " Cell2 : " + sdv_cell2 + " Cell3 : " + sdv_cell3);
+    //   Serial.println("");
+    //   Serial.println("");
+
+
+    // }
+    last_print_time = millis();
```

Comment:
Moving this to `battery_buzzer.cpp::loop` would allow making this variable non-extern. Moreover, the entire name `last_print_time` seems not to accurately reflect the usage of this variable.